pkg/controller/dynamicupgradeableconditionscontroller.go

```diff
@@ -0,0 +1,300 @@
+	infFact := dynamicinformer.NewDynamicSharedInformerFactory(client, 0)
+
+	clusterExtensionGVR := schema.GroupVersionResource{
+		Group:    ocv1alpha1.ClusterExtensionGVK.Group,
+		Version:  ocv1alpha1.ClusterExtensionGVK.Version,
+		Resource: ocv1alpha1.ClusterExtensionGVK.Kind,
+	}
+
+	inf := infFact.ForResource(clusterExtensionGVR)
+
+	c := &dynamicUpgradeableConditionController{
+		informer:       inf,
+		client:         client,
+		kubeclient:     kubeclient,
+		name:           name,
+		operatorClient: operatorClient,
+		prefixes:       prefixes,
+	}
+
+	inf.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.handleClusterExtension,
+		UpdateFunc: func(oldObj, newObj interface{}) { c.handleClusterExtension(newObj) },
+		DeleteFunc: c.handleClusterExtension,
+	})
+
+	return factory.New().WithInformers(inf.Informer()).ToController(name, eventRecorder)
+}
+
+func (c *dynamicUpgradeableConditionController) handleClusterExtension(obj interface{}) {
+	_, ok := obj.(*ocv1alpha1.ClusterExtension)
+	if !ok {
+		log.Println("Error: could not cast to ClusterExtension object")
+		return
+	}
+
+	c.syncIncompatibleOperators()
+}
+
+func (c *dynamicUpgradeableConditionController) syncIncompatibleOperators() {
+	current, err := getCurrentRelease()
+	if err != nil {
+		log.Printf("Error getting current OCP release: %v", err)
+		return
+	}
+
+	if current == nil {
+		// Note: This shouldn't happen
+		err = fmt.Errorf("failed to determine current OpenShift Y-stream release")
+		log.Printf("Error processing secrets: %v", err)
+		return
+	}
+
+	next, err := nextY(*current)
+	if err != nil {
+		log.Printf("Error finding next OCP Y-stream release: %v", err)
+		return
+	}
+
+	var incompatibleOperators []string
+
+	ceList, err := c.informer.Lister().List(nil)
+	if err != nil {
+		log.Printf("Error listing cluster extensions: %v", err)
+		return
+	}
+
+	// Get all ClusterExtensions incompatible with next Y-stream
+	for _, obj := range ceList {
+		ce, ok := obj.(*ocv1alpha1.ClusterExtension)
+		if !ok {
+			log.Printf("Error: could not cast to ClusterExtension object")
+			return
+		}
+		store := buildHelmStore(ce, c.kubeclient.CoreV1().Secrets(ce.Namespace))
+
+		rel, _ := store.Deployed(ce.Name)
+		props, err := propertyListFromPropertiesAnnotation(rel.Chart.Metadata.Annotations["olm.properties"])
+		if err != nil {
+			log.Printf("Error converting olm.properties: %v", err)
+			return
+		}
+		for _, p := range props {
+			if p.Type == MaxOpenShiftVersionProperty {
+				version := string(p.Value)
+				maxOCPVersion, err := toSemver(version)
+				if err != nil {
+					log.Printf("Error converting to semver for version %s: %v", version, err)
+					continue
+				}
+				if maxOCPVersion == nil || maxOCPVersion.GTE(next) {
+					// All good
+				} else {
+					// Incompatible
+					name := rel.Labels[BundleNameKey]
+					incompatibleOperators = append(incompatibleOperators, name)
```

Comment:
I know this would differ from OLMv0, but it might be nice to include the name of the ClusterExtension somehow in the error message, which would give users a breadcrumb to the object they need to change in order to resolve this condition.

Reply:
agreed - let's create a new ticket to track that addition and I can create a follow up PR